File: Assets/Scripts/Core/Rules/MoveOption.cs
namespace Parkiller.Core
{
    public enum MoveKind
    {
        ExitYard,
        TrackMove,
        CorridorMove,
        FinishMove
    }

    public class MoveOption
    {
        public Piece Piece;
        public MoveKind Kind;
        public int ResultingTrackPosition = -1;
        public int ResultingCorridorPosition = -1;
    }

    public class MoveResult
    {
        public Piece MovedPiece;
        public Piece CapturedPiece;
        public bool PieceFinished;
    }
}

File: src/core/rules/moveOption.ts
import type { Piece } from '../pieces/piece'

export type MoveKind = 'ExitYard' | 'TrackMove' | 'CorridorMove' | 'FinishMove'

export interface MoveOption {
  piece: Piece
  kind: MoveKind
  resultingTrackPosition: number
  resultingCorridorPosition: number
}

export interface MoveResult {
  movedPiece: Piece
  capturedPiece: Piece | null
  pieceFinished: boolean
}
